src/plugins/gui/providers/mtpelerinProvider.ts

```diff
@@ -191,6 +191,11 @@ const allowedCountryCodes: { [code: string]: boolean } = {
   VA: true // Vatican City
 }
 
+const asApiKeys = asObject({
+  apiKey: asString,
+  referralCode: asOptional(asString)
```

Comment:
Would it make more sense to do `asOptional(asString, 'edgewallet')` here? The way you have it is fine too.

Reply:
I'd rather clean it and add business logic in the plugin